cetmix_tower_server/models/cx_tower_reference_mixin.py

```diff
@@ -186,11 +186,14 @@ def _get_copied_name(self):
         copy_name = _("%(name)s (copy)", name=original_name)
 
         counter = 1
-        while self.search_count([("name", "=", copy_name)]) > 0:
-            counter += 1
-            copy_name = _(
-                "%(name)s (copy %(number)s)", name=original_name, number=str(counter)
-            )
+        if not self.env.context.get("skip_copy_action"):
+            while self.search_count([("name", "=", copy_name)]) > 0:
+                counter += 1
+                copy_name = _(
+                    "%(name)s (copy %(number)s)",
+                    name=original_name,
+                    number=str(counter),
+                )
```

Comment:
you can reduce code nesting, which will improve its readability:

```python
if self.env.context.get("skip_copy_action"):
    return copy_name

while .......
```

please, check it.

And please add comments as to why we should use this context and only then execute the loop